common/src/chain/config/mod.rs

```diff
@@ -94,6 +95,7 @@ pub struct ChainConfig {
     genesis_block: Arc<WithId<Genesis>>,
     max_future_block_time_offset: Duration,
     version: SemVer,
+    user_agent: UserAgent,
```

Comment:
Isn't it more appropriate to put this in p2p config?

Reply:
Yes, probably. It was easier to add to `ChainConfig` and `version` was already there. Moved to p2p config.

Reply:
Should we move `version` to `P2pConfig` for consistency? It's only used in p2p.

Reply:
Perhaps, yes. I'm thinking version counts as a consensus parameter. That's why it should be here. But agent isn't. All subsystems need to know the version of the node to make decisions.